### DISCUSSION TITLE
Issues Removing Server from Cronicle (v0.9.39) Due to Authentication Failure

### DISCUSSION THREAD

**Discussion by abx-luishau** [jhuckaby/Cronicle#750] [Q&A] [answered]
Hello,

I am currently facing an issue with my Cronicle setup where one of the servers in the cluster (ip-10-64-10-80) is repeatedly logging authentication failures and has been removed from the cluster. The specific errors in the logs are:

- Authentication failure, cannot add slave: ip-10-64-10-80 (Secret Keys do not match.)
- Server removed from cluster: ip-10-64-10-80

I have reviewed the Wiki and attempted to address this by inspecting the global servers list via the following command:

`/opt/cronicle/bin/storage-cli.js get global/servers/0`

The output confirmed the presence of the problematic server alongside my intended master server:

```
{
	"type": "list_page",
	"items": [
		{
			"hostname": "ip-10-64-8-80",
			"ip": "10.64.8.80"
		},
		{
			"hostname": "ip-10-64-10-80",
			"ip": "10.64.10.80"
		}
	]
}
```

I would like to permanently remove `ip-10-64-10-80` from the cluster and continue with `ip-10-64-8-80` as my sole master server. Could someone guide me on how to correctly remove the server from the configuration to prevent any future errors? Thanks in advance

**Comment by jhuckaby** [comment, marked as answer]
The problem is with your config files, not the server database.  Each Cronicle server has a config file located here (by default):

```
/opt/cronicle/conf/config.json
```

These files all have a `secret_key` property, and they must always match to be part of a cluster.  It sounds like your worker server (`ip-10-64-10-80`) has a mismatched secret key vs. your master server (`ip-10-64-8-80 `).

To fix the problem, edit the config files on both machines, and make sure their `secret_key` properties match exactly.

To remove the server entirely, you can use the Cronicle UI on your master server.  Login, then click on the "Admin" tab at the top of the page, then click the "Servers" tab along the side.  Find the server you want to remove, and click the "Remove" link on the right side.  Then, you can shut down and remove the Cronicle software from the server.

Good luck.

**Reply by abx-luishau** [reply]
Thank you, @jhuckaby for your prompt response. My intention is to remove the host `ip-10-64-10-80` from the cluster, and make the host `ip-10-64-8-80` a standalone server, as I have lost my connectivity to access the server `ip-10-64-10-80`. In the Cronicle UI, there is only one hostname listed:
![CleanShot 2024-05-06 at 09 18 48](https://github.com/jhuckaby/Cronicle/assets/163954556/63e1c129-d4fb-4e2e-b2f8-871e2391156c)

Are there any ways to make my master server stop looking for its slave server?

**Comment by jhuckaby** [comment]
If the server isn't listed in the UI, it has already been removed, and is not part of the cluster any longer.
